generator/internal/stub_factory_generator.cc

```diff
@@ -51,9 +51,13 @@ Status StubFactoryGenerator::GenerateHeader() {
   auto result = HeaderOpenNamespaces(NamespaceType::kInternal);
   if (!result.ok()) return result;
 
+  HeaderPrint(  // clang-format off
+    "internal::Options Resolve$service_name$Options(internal::Options options);\n\n");
```

Comment:
I think this should be `Default$service_name$Options()`, and the argument should have a default value of `= {}`. This lets us either fill in an existing options with the defaults, or construct a new object fully filled in with defaults.